src/toil/lib/pipes.py

```diff
@@ -117,7 +117,12 @@ def __exit__(self, exc_type: Optional[str], exc_val: Optional[str], exc_tb: Opti
             # thread. To cover the small window before the reader takes over we also close it here.
             if self.readable_fh is not None:
                 # Close the file handle. The reader thread must be dead now.
-                os.close(self.readable_fh)
+                try:
+                    os.close(self.readable_fh)
+                except OSError as e:
+                    # OSError: [Errno 9] Bad file descriptor implies this file handle is already closed
+                    if not e.errno == 9:
```

Comment:
This should be `errno.EBADF` and not a magic 9.
```suggestion
                    if not e.errno == errno.EBADF:
```